front/lib/resources/key_resource.ts

```diff
@@ -163,6 +166,16 @@ export class KeyResource extends BaseResource<KeyModel> {
     );
   }
 
+  async updateSecret() {
```

Comment:
`nit` wording is not correct here. Also I'd not make this a class instance. Rotating a key needs to be somehow atomic. Here is the key might be used in many different places. 

```suggestion
  async rotateSecret() {
```